### PR TITLE
upgrade autorest.csharp version and refresh RPs

### LINKED ISSUE #29310: [LLC] upgrade autorest.csharp version
Current autorest.csharp has released an new version, we need to upgrade autorest.csharp version:
1. upgrade the autorest.csharp version
2. update the convenience methods to resolve build failure.

### PR DESCRIPTION
**Pull request by chunyu3**
# Contributing to the Azure SDK
Fix https://github.com/Azure/azure-sdk-for-net/issues/29310

- Upgrade autorest.csharp version to 3.0.0-beta.20220613.3
- fresh all the PRs

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

File: sdk/cognitivelanguage/Azure.AI.Language.Conversations/src/Generated/ConversationAnalysisClient.cs
Comment (heaths) [inline sdk/cognitivelanguage/Azure.AI.Language.Conversations/src/Generated/ConversationAnalysisClient.cs:384]: These are breaking. The LROs eventually return a result (except DELETEs if returning a 204 eventually) but use `Operation-Location` header to refer to a status monitor per guidelines: https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md#long-running-operations-with-status-monitor. Unfortunately in this case, they don't use the old method of returning both a 200 from the LRO to show the response model + the 202 to provide the `Operation-Location` header. Maybe we need to do this in the short term, but @mikekistler isn't crazy about that.

**Review by heaths — DISMISSED** [review]

**Comment by heaths** [comment]
All the CLU LROs should return `Operation<BinaryData>`. See #29331 and an email I just replied to for more information.

The transforms I added in https://github.com/heaths/azure-sdk-for-net/blob/f0273d93b5e8d6e53a20361770cf66329fc27116/sdk/cognitivelanguage/Azure.AI.Language.Conversations/src/autorest.md will work for the CLU 1.0 SDK I have in a separate release branch, but in the `main` branch we have the 1.1 beta SDK that has a few more LROs in analyzeconversation.json that will need to be similarly transformed. I could try to do this later today, but my focus has been on the CLU 1.0 GA now, which was always planned to ship from a release branch.

If critical, perhaps in `main` we just remove the project from building. See _sdk/keyvault/service.projects_ for how to do that. Copy the file and update the project reference if absolutely necessary to unblock mgmt.

/cc @annelo-msft @pallavit

**Comment by lirenhe** [comment]
Thanks @heaths.  

@annelo-msft,  I did a check of the build failure and there are 2 services impacted.
Azure.AI.Language.QuestionAnswering
Azure.AI.Language.Conversations

mgmt. Plane team has some urgent ask that want us to unblock them, so shall we take the approach suggested by Healths?

_If critical, perhaps in main we just remove the project from building. See sdk/keyvault/service.projects for how to do that. Copy the file and update the project reference if absolutely necessary to unblock mgmt._

**Comment by annelo-msft** [comment]
Hi @lirenhe, yes, please use the approach Heath recommended.

**Comment by pshao25** [comment]
> All the CLU LROs should return `Operation<BinaryData>`. See #29331 and an email I just replied to for more information.
> 
> The transforms I added in https://github.com/heaths/azure-sdk-for-net/blob/f0273d93b5e8d6e53a20361770cf66329fc27116/sdk/cognitivelanguage/Azure.AI.Language.Conversations/src/autorest.md will work for the CLU 1.0 SDK I have in a separate release branch, but in the `main` branch we have the 1.1 beta SDK that has a few more LROs in analyzeconversation.json that will need to be similarly transformed. I could try to do this later today, but my focus has been on the CLU 1.0 GA now, which was always planned to ship from a release branch.
> 
> If critical, perhaps in `main` we just remove the project from building. See _sdk/keyvault/service.projects_ for how to do that. Copy the file and update the project reference if absolutely necessary to unblock mgmt.
> 
> /cc @annelo-msft @pallavit

@heaths we cannot exculde all the projects in the same service folder, with error:
Error : No Projects found with patttern [..\sdk\cognitivelanguage/**\*.csproj], please make sure you have passed in the correct ServiceDirectory.

Do you have other solutions?

**Comment by lirenhe** [comment]
@heaths,  as the original approach has some limitation, we tried to remove this global check and make most of the CI work. 
https://github.com/Azure/azure-sdk-for-net/pull/29314/commits/c52be68e67314a121e2038b15fee88c6a8c16ecc

As this change would impact all packages, we are not sure whether this is the right approach to work around this problem.
So please let us know if you are OK with this approach or you have other suggestions.  Thanks

**Comment by annelo-msft** [comment]
Heath has updated the main branch in a way we believe will unblock the autorest.csharp version upgrade in https://github.com/Azure/azure-sdk-for-net/pull/29359.  '

@heaths, can you confirm that once the merge conflicts have been resolved, this PR should be able to be merged?  Would you be willing to remove your requested changes to unblock the merge?  Thanks!

File: eng/service.proj
Comment (pshao25) [inline eng/service.proj:45]: A space here.

**Review by pshao25 — APPROVED** [review]

**Review by archerzz — APPROVED** [review]

File: sdk/cognitivelanguage/Azure.AI.Language.Conversations/src/Generated/ConversationAnalysisClient.cs
Comment (chunyu3) [reply sdk/cognitivelanguage/Azure.AI.Language.Conversations/src/Generated/ConversationAnalysisClient.cs]: resolved

**Pull request merged**
